libs/langchain/langchain/llms/google_palm.py

```diff
@@ -117,31 +173,164 @@ def _generate(
+        if stop:
+            generation_config["stop_sequences"] = stop
+        for stream_resp in completion_with_retry(
+            self,
+            prompt,
+            stream=True,
+            is_gemini=True,
+            run_manager=run_manager,
+            generation_config=generation_config,
+            **kwargs,
+        ):
+            chunk = GenerationChunk(text=stream_resp.text)
+            yield chunk
+            if run_manager:
+                run_manager.on_llm_new_token(
+                    stream_resp.text,
+                    chunk=chunk,
+                    verbose=self.verbose,
+                )
+
+    def invoke(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> str:
+        generation_config = {
+            "stop_sequences": stop,
+            "temperature": self.temperature,
+            "top_p": self.top_p,
+            "top_k": self.top_k,
+            "max_output_tokens": self.max_output_tokens,
+            "candidate_count": self.n,
+        }
+        if isinstance(input, str):
+            return super().invoke(input, config, stop=stop, **generation_config)
+        elif isinstance(input, StringPromptValue):
+            return super().invoke(
+                input.to_string(), config, stop=stop, **generation_config
+            )
+        elif isinstance(input, list):
+            if "vision" not in self.model_name:
+                raise ValueError(
+                    f"Multi-modal input is not supported by {self.model_name} model!"
+                )
+            first_message = input[0]
+            if len(input) > 1 or not isinstance(first_message, BaseMessage):
+                raise ValueError(
+                    "Multi-modal model expects only a single message as a input!"
+                )
+        else:
+            raise ValueError(
+                f"Invalid input type {type(input)}. "
+                "Must be a PromptValue, str, or list of BaseMessages."
             )
 
-            prompt_generations = []
-            for candidate in completion.candidates:
-                raw_text = candidate["output"]
-                stripped_text = _strip_erroneous_leading_spaces(raw_text)
-                prompt_generations.append(Generation(text=stripped_text))
-            generations.append(prompt_generations)
+        messages = []
+        for content in first_message.content:
+            if not isinstance(content, Dict):
+                raise ValueError(
+                    f"Message's content is expected to be a dict, got {type(content)}!"
+                )
+            if content["type"] == "text":
+                messages.append(content["text"])
+            elif content["type"] == "image_url":
+                path = content["image_url"]["url"]
+                mime_type = content["image_url"]["mime_type"]
```

Comment:
This one usually isn't provided in our current flows -> would be good to make optional

Reply:
done